### PR TITLE
Like increment

### PR DESCRIPTION
**Pull request by tsrahm**
I don't think we need the componentWillReceiveProps() lifecycle, as we're doing a setState in the clickHandler, but please confirm.

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -192,7 +192,7 @@ app.put('/definitions/:id', checkJwt, function (req, res) {
 
 // PUT to existing definition -- like
 app.put('/definitions/:id/likes', checkJwt, function (req, res) {
-  acronymsModel.likeDefinition(req.params.id, req.user.sub, db, (err) => {
+  acronymsModel.likeDefinition(ObjectId.createFromHexString(req.params.id), req.user.sub, db, (err) => {
```
Comment (NFabrizio) [inline server/server.js]: Good ol Mongo stuff.

File: src/AcronymLike.js [hunk 2]
```diff
@@ -39,15 +39,8 @@ class AcronymLike extends React.Component {
...
-  componentWillReceiveProps(nextProps) {
-    const liked = this.isLiked(nextProps);
-    this.setState({
-      liked
-    });
-  }
-
   clickHandler() {
-    if (!this.props.auth || !this.props.auth.userProfile) {
+    if (!this.props.auth || !this.props.auth.userProfile || this.state.liked) {
```
Comment (NFabrizio) [inline src/AcronymLike.js]: Nice and simple.

File: src/AcronymLike.js [hunk 3]
```diff
@@ -39,15 +39,8 @@ class AcronymLike extends React.Component {
     });
   }
 
-  componentWillReceiveProps(nextProps) {
```
[the remark this answers is not in the document]
Comment (nlkluth) [reply src/AcronymLike.js]: currently, homepage shows all thumbs as black until you like + refresh then they are all blue

[the remark this answers is not in the document]
Comment (tsrahm) [reply src/AcronymLike.js]: Yeah, so I realized after I removed componentWillReceiveProps why it was there, and that's when I decided to do the refactor, which includes (some of) these changes. I didn't cancel this PR, cuz I thought you might not approve the refactor PR.

**Pull request merged**